packages/patternfly-4/react-core/src/components/Select/SelectToggle.js

```diff
@@ -0,0 +1,142 @@
+import React, { Component } from 'react';
+import styles from '@patternfly/patternfly-next/components/Select/select.css';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+import { CaretDownIcon } from '@patternfly/react-icons';
+
+const propTypes = {
+  /** HTML ID of dropdown toggle */
+  id: PropTypes.string.isRequired,
+  /** Anything which can be rendered as dropdown toggle */
+  children: PropTypes.node,
+  /** Classes applied to root element of dropdown toggle */
+  className: PropTypes.string,
+  /** Flag to indicate if select is expanded */
+  isExpanded: PropTypes.bool,
+  /** Callback called when toggle is clicked */
+  onToggle: PropTypes.func,
+  /** Element which wraps toggle */
+  parentRef: PropTypes.any,
+  /** Forces focus state */
+  isFocused: PropTypes.bool,
+  /** Forces hover state */
+  isHovered: PropTypes.bool,
+  /** Forces active state */
+  isActive: PropTypes.bool,
+  /** Display the toggle with no border or background */
+  isPlain: PropTypes.bool,
+  /** Type of the toggle button, defaults to 'button' */
+  type: PropTypes.string,
+  /** The icon to display for the toggle. Defaults to CaretDownIcon. Set to null to not show an icon. */
+  iconComponent: PropTypes.func,
+  /** Additional props are spread to the container <button> */
+  '': PropTypes.any
+};
+
+const defaultProps = {
+  children: null,
+  className: '',
+  isExpanded: false,
+  parentRef: null,
+  isFocused: false,
+  isHovered: false,
+  isActive: false,
+  isPlain: false,
+  iconComponent: CaretDownIcon,
+  onToggle: Function.prototype
+};
+
+class SelectToggle extends Component {
+  componentDidMount = () => {
```

Comment:
these can be written as `componentDidMount()` and `componentWillMount()` There isn't a bunch of overhead, but making these instance properties will use more memory and have a slower instantiation.